modules/magnolia/src/test/scala/pureconfig/module/magnolia/DerivationModesSuite.scala

```diff
@@ -58,4 +83,12 @@ class DerivationModesSuite extends BaseSuite {
     ConfigReader[Entity].from(conf.root) shouldBe Right(person)
     ConfigWriter[Entity].to(person) shouldBe conf.root()
   }
+
+  it should "use existing reader and writer instances when they exist" in {
```

Comment:
Can you also include this test in the `DerivationModesSuite` of [`pureconfig-tests`](https://github.com/pureconfig/pureconfig/blob/master/tests/src/test/scala/pureconfig/DerivationModesSuite.scala)? We're currently not reusing the base test suite for the generic derivation modules, but I think a test like this is also relevant to prevent a regression from being introduced in the shapeless-based derivation.

Reply:
@jcazevedo Included!